cc/testing/evaluation/algorithms_with_insufficient_noise.cc

```diff
@@ -0,0 +1,187 @@
+ 
+//
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// #include "bounded-sum-with-insufficient-noise.h"
+
+#include "algorithms_with_insufficient_noise.h"
+
+#include <chrono>
+#include <ctime>
+#include <fstream>
+#include <iterator>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "absl/memory/memory.h"
+#include "absl/random/distributions.h"
+
+#include "algorithms/algorithm.h"
+#include "algorithms/bounded-sum.h"
+#include "algorithms/bounded-mean.h"
+#include "algorithms/count.h"
+#include "algorithms/numerical-mechanisms.h"
+#include "algorithms/numerical-mechanisms-testing.h"
+#include "algorithms/util.h"
+#include "base/statusor.h"
+#include "proto/data.pb.h"
+#include "testing/sequence.h"
+#include "testing/stochastic_tester.h"
+
+namespace differential_privacy {
+
+namespace testing {
+
+// Runs the Stochastic Tester on a continuous range of three algorithms (Count, 
+// BoundedSum, and BoundedMean) in which the privacy protection claimed exceeds 
+// the amount of noise applied. Sends results of each run to an output file.
+
+bool RunStochasticTester(std::string algorithm, double ratio, double num_datasets,
+  double num_samples_per_histogram) {
+
+  auto sequence = absl::make_unique<HaltonSequence<double>>(
+    DefaultDatasetSize(), true, DefaultDataScale(), DefaultDataOffset());
+
+  if (algorithm == "count") {
```

Comment:
Going to follow up with this offline, will explain why I decided to make it all one function!!

Reply:
For what it's worth, I agree that having this method call three almost completely separate code blocks isn't great. Instead of having a string (or enum) argument that is used to select the construction behavior, I'd suggest passing in a constructed object instead.